### DISCUSSION TITLE
Designing & Planning Discussion

### DISCUSSION THREAD

**Discussion by sanket1211996** [airavata-courses/SuperNova#12] [Ideas]
I have created this first draft of napkin diagram. Let me know what can we improve or add in this. 

[Napkin Diagram](https://github.com/airavata-courses/SuperNova/wiki/Napkin-Diagram)

**Comment by richajha012** [comment]
we can add like "if yes then this is the right platform for you" because ultimately we are giving that functionality so we can show that right ?? Let me know If I'm right

**Reply by sanket1211996** [reply]
Yes, thats a good point. But I actually had added similar line on the central part.

**Reply by richajha012** [reply]
yes right, Its looks perfect

**Comment by sumedhsalvi** [comment]
The napkin diagram looks perfect. Also, we need to draw a sequence diagram of architecture in the Napkin style.

**Reply by sanket1211996** [reply]
Sure

**Comment by sumedhsalvi** [comment]
Minor typo correction - missing apostrophe for word "earths" in the statement 'Get Eagle of earth weather, with weather data'

**Reply by sanket1211996** [reply]
Sure, I will update this.

**Reply by sanket1211996** [reply]
I have updated the PR. Verify and merge.

**Reply by sumedhsalvi** [reply]
I have verified and merged the changes. You can close the issue

**Comment by sanket1211996** [comment]
I have added the user dashboard wireframe diagram. #27 . Take a look and let me know if any changes needed.

**Reply by sanket1211996** [reply]
https://github.com/airavata-courses/SuperNova/wiki/User-Dashboard-Wireframe

**Comment by sumedhsalvi** [comment]
I have created a Project Planning diagram. #29 Please take a look and let me know if any changes are required.

**Reply by sanket1211996** [reply]
I have reviewed the PR. Good Job. Merged the PR. Do add it to wiki pages. @sumedhsalvi

**Reply by sumedhsalvi** [reply]
Added to Wiki Page. 
https://github.com/airavata-courses/SuperNova/wiki/Project-Planning-Diagram

**Comment by sanket1211996** [comment]
USER API DATABASE DESIGN CHANGES
session-info
PK(id) AUTO_INC ,  FK(userID) NOT NULL,  radStation dataTime sessionTime
1 ABC T T
1 ABC T T
2. BCD T T

user-info
PK(id) AUTO_INC, email NOT_NULL
1. sanket 
2. ricjha

**Reply by richajha012** [reply]
do we need to make user-id as auto increment?

**Reply by sanket1211996** [reply]
I have updated the above design.

**Reply by richajha012** [reply]
okay

**Reply by richajha012** [reply]
create table SessionInfo(
   id INT NOT NULL AUTO_INCREMENT,
   user_id INT NOT NULL,
   radStation VARCHAR(250),
   dataTime Date,
   SessionTime TIMESTAMP,
   PRIMARY KEY ( user_id ),
   FOREIGN KEY (user_id) REFERENCES UserInfo(user_id)
);

create table UserInfo(
   user_id INT NOT NULL AUTO_INCREMENT,
   user_email VARCHAR(250) NOT NULL,
   PRIMARY KEY ( user_id )
);


These are the updated ones. Let me know if you guys think it fine to go-ahead

**Reply by sanket1211996** [reply]
if field is primary key, it is not needed to do NOT_NULL its by default NOT_NULL

**Reply by richajha012** [reply]
yea missed it . Thanks

**Reply by richajha012** [reply]
create table SessionInfo(
id INT  AUTO_INCREMENT,
user_id INT NOT NULL,
radStation VARCHAR(250),
dataTime Date,
SessionTime TIMESTAMP,
PRIMARY KEY ( id ),
FOREIGN KEY (user_id) REFERENCES UserInfo(user_id)
);

create table UserInfo(
user_id INT AUTO_INCREMENT,
user_email VARCHAR(250) NOT NULL,
PRIMARY KEY ( user_id )
);

**Reply by sanket1211996** [reply]
UserInfo table should have id field not user_id. Reference table should have named the Foreign key as user_id.

**Reply by sanket1211996** [reply]
userID

**Reply by richajha012** [reply]
I have Created Pull requests for user DB scripts and user DB diagrams for issues #1, #34, Please review and merge.

**Comment by sanket1211996** [comment]
I am tagging all the issue with project 1 label. Hence forth all the new issues will be tagged with project label. @richajha012 @sumedhsalvi

**Reply by sumedhsalvi** [reply]
Okay. Noted!

**Comment by sanket1211996** [comment]
I have design a high level architecture, as discuss and proposed design. Do review and merge. @sumedhsalvi  @richajha012

**Reply by sumedhsalvi** [reply]
I have reviewed and requested small corrections.
Please check.

**Reply by sanket1211996** [reply]
Good catch, modified the needful. Please do review and merge.

**Reply by sumedhsalvi** [reply]
I have reviewed and merged the changes.
